packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts

```diff
@@ -33,15 +36,22 @@ export class AppSyncModelJavaVisitor<
   protected additionalPackages: Set<string> = new Set();
   protected usingAuth: boolean = false;
 
+  // We override lazy generation to also only generate lazy models if data store is disabled
+  protected override isGenerateModelsForLazyLoadAndCustomSelectionSet(): boolean {
+    return super.isGenerateModelsForLazyLoadAndCustomSelectionSet() && !(this.config.isDataStoreEnabled ?? false);
+  }
```

Comment:
nit: maybe `APIOnly` somewhere in the naming of this computed property? `isGenerateModelsForLazyLoadAndCustomSelectionSetAPIOnly` or `isGenerateAPIModelsForLazyLoadAndCustomSelectionSet`

Reply:
Problem in that is that this is an override.